Core/Core/Inbox/ComposeMessage/ViewModel/ComposeMessageViewModel.swift

```diff
@@ -35,29 +33,56 @@ class ComposeMessageViewModel: ObservableObject {
         // && (attachments.isEmpty || attachments.allSatisfy({ $0.isUploaded }))
 
     }
+    public var isReply: Bool {
+        conversation != nil
+    }
 
     // MARK: - Inputs
     public let sendButtonDidTap = PassthroughRelay<WeakViewController>()
     public let cancelButtonDidTap = PassthroughRelay<WeakViewController>()
     public let addRecipientButtonDidTap = PassthroughRelay<WeakViewController>()
-    public let selectedRecipient = CurrentValueRelay<SearchRecipient?>(nil)
+    public let selectedRecipient = CurrentValueRelay<Recipient?>(nil)
 
     // MARK: - Inputs / Outputs
     @Published public var sendIndividual: Bool = false
     @Published public var bodyText: String = ""
     @Published public var subject: String = ""
     @Published public var selectedContext: RecipientContext?
+    @Published public var conversation: Conversation?
 
     // MARK: - Private
     private var subscriptions = Set<AnyCancellable>()
     private let interactor: ComposeMessageInteractor
     private let router: Router
     private let scheduler: AnySchedulerOf<DispatchQueue>
 
-    public init(router: Router, interactor: ComposeMessageInteractor, scheduler: AnySchedulerOf<DispatchQueue> = .main) {
+    public init(router: Router, conversation: Conversation? = nil, author: String? = nil, interactor: ComposeMessageInteractor, scheduler: AnySchedulerOf<DispatchQueue> = .main) {
         self.interactor = interactor
         self.router = router
         self.scheduler = scheduler
+        self.conversation = conversation
+
+        if let conversation {
+            self.subject = conversation.subject
+            let rawContext = conversation.contextCode?.split(separator: "_") ?? []
+            switch rawContext[0] {
+            case "course":
+                self.selectedContext = RecipientContext(name: conversation.contextName ?? "", context: Context.course(String(rawContext[1])))
+            case "group":
+                self.selectedContext = RecipientContext(name: conversation.contextName ?? "", context: Context.group(String(rawContext[1])))
+            default:
+                self.selectedContext = RecipientContext(name: conversation.contextName ?? "", context: Context.course(String(rawContext[1])))
+            }
```

Comment:
I think we could use `Context(canvasContextID: String)` to make this snippet more simple.